packages/firestore/src/remote/remote_store.ts

```diff
@@ -155,25 +155,26 @@ export class RemoteStore {
 
   /**
    * Updates our OnlineState to the new state, updating local state
-   * and notifying the onlineStateHandler as appropriate.
+   * and notifying the onlineStateHandler as appropriate. Idempotent.
    */
-  private updateOnlineState(newState: OnlineState) {
+  private updateOnlineState(newState: OnlineState): void {
     if (newState === OnlineState.Healthy) {
```

Comment:
FWIW, it is safe (and makes the "Idempotent." part of the method comment more correct) to move this inside the `if (newState !== this.watchStreamOnlineState) { ... }` check, so that would be my preference.  

And at that point, if you felt like it, you could flip the condition and make it an early return if you felt like it (note though that we generally avoid early return except for these sorts of "there's nothing to do, so just return immediately" cases).

Reply:
Moved it inside.